dockers/docker-snmp-sv2/Dockerfile.j2

```diff
@@ -1,48 +1,40 @@
 FROM docker-base
 
-COPY \
+COPY [ \
 {% for deb in docker_snmp_sv2_debs.split(' ') -%}
-debs/{{ deb }}{{' '}}
-{%- endfor -%}
-debs/
+    "debs/{{ deb }}",
+{%- endfor %} \
+    "/debs/"]
 
-COPY python-wheels /python-wheels
+COPY python-wheels/sswsdk-*-py3-*.whl /python-wheels/
+COPY python-wheels/asyncsnmp-*-py3-*.whl /python-wheels/
 
 # enable -O for all Python calls
 ENV PYTHONOPTIMIZE 1
 
-## Pre-install the fundamental packages
-## Install Python SSWSDK (SNMP subagent dependency)
-## Install SNMP subagent
-## Clean up
-RUN apt-get update && apt-get install -y libmysqlclient-dev libmysqld-dev libperl-dev libpci-dev libpci3 libsensors4 libsensors4-dev libwrap0-dev
-
-# Dependencies for sonic-cfggen                                                                                                                                                           
-RUN apt-get install -y python-lxml python-jinja2 python-netaddr python-ipaddr python-yaml 
-
-RUN dpkg -i \
-{% for deb in docker_snmp_sv2_debs.split(' ') -%}
-debs/{{ deb }}{{' '}}
-{%- endfor %}
-
-RUN rm -rf /debs
-
+# install supervisor
+# install libsnmp30 dependencies
+# install dependencies for sonic-cfggen
+# install libpython3.6-dev and pip dependencies
+# TODO: remove libpython3.6-dev, its and pip's dependencies if we can get pip3 directly
 # install subagent
-RUN apt-get -y install build-essential wget libssl-dev openssl supervisor && \
-    rm -rf /var/lib/apt/lists/* && \
-    wget https://www.python.org/ftp/python/3.5.2/Python-3.5.2.tgz && \
-    tar xvf Python-3.5.2.tgz && cd Python-3.5.2 && \
-    ./configure --without-doc-strings --prefix=/usr --without-pymalloc --enable-shared && \
-    make && make install && \
-    ldconfig && \
-    cd .. && rm -rf Python-3.5.2 && rm Python-3.5.2.tgz && \
-    pip3 install --no-cache-dir /python-wheels/*py3*.whl hiredis && \
+# clean up
```

Comment:
You are right. Both options have the same effect. The merging option is kind of manual 'docker squash', which remove the unnecessary intermediate layer storage from docker image's point of view. In future when 'docker squash' becomes mature, we can prevent this situation.

---
In reply to: [99679265](https://github.com/Azure/sonic-buildimage/pull/245#discussion_r99679265) [](ancestors = 99679265)